mozetl/maudau/maudau.py

```diff
@@ -0,0 +1,214 @@
+"""
+Calculate Firefox monthly- and daily-active users.
+"""
+import csv
+import datetime as DT
+import logging
+import os
+
+from pyspark.sql import SparkSession
+import boto3
+import botocore
+import click
+from boto3.s3.transfer import S3Transfer
+from moztelemetry.standards import (
+    filter_date_range,
+    count_distinct_clientids
+)
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+DEVELOPMENT = False
+
+STORAGE_BUCKET = "net-mozaws-prod-us-west-2-pipeline-analysis"
+STORAGE_SUB_DIR = 'spenrose/maudau' if DEVELOPMENT else 'mreid/maudau'
+MAUDAU_ROLLUP_BASENAME = "engagement_ratio.csv"
+MAUDAU_SNAPSHOT_TEMPLATE = "engagement_ratio.{}.csv"
+DASHBOARD_BUCKET = "net-mozaws-prod-metrics-data"
+DASHBOARD_KEY = "firefox-dashboard/{}".format(MAUDAU_ROLLUP_BASENAME)
+
+today = DT.date.today()
+yesterday = today - DT.timedelta(1)
+ACTIVITY_SUBMISSION_LAG = DT.timedelta(10)
+# MONTH as in "monthly active"
+MONTH = 28
+# Recency in days of data which should be re-counted.
+INSTABLE_WINDOW = 30
+
+
+def format_as_submission_date(isoformat_date):
+    return isoformat_date.replace('-', '')[:8]
+
+
+def format_spark_path(bucket, prefix):
+    return "s3://{}/{}".format(bucket, prefix)
+
+
+def get_rollup_s3_paths(basename):
+    return (STORAGE_BUCKET, os.path.join(STORAGE_SUB_DIR, basename))
+
+
+def generate_filter_parameters(end_date, days_back):
```

Comment:
This might look better inline with the parent function. I would avoid returning more than 2 values in a function (a status, value tuple is one case where I think this style is somewhat appropriate).